microservices/Keli-Katheya-User-Registration/src/main/java/com/kelikatheya/microservices/userRegistration/services/impls/UserServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.kelikatheya.microservices.userRegistration.services.impls;
+
+import com.kelikatheya.microservices.userRegistration.entity.UserEntity;
+import com.kelikatheya.microservices.userRegistration.entity.UserRepository;
+import com.kelikatheya.microservices.userRegistration.model.UserModel;
+import com.kelikatheya.microservices.userRegistration.requestModels.UserRequest;
+import com.kelikatheya.microservices.userRegistration.services.UserService;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.convention.MatchingStrategies;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Autowired
+    BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Override
+    public UserModel registerUser(UserRequest userRequest) {
+
+        ModelMapper modelMapper = new ModelMapper();
+        modelMapper.getConfiguration().setMatchingStrategy(MatchingStrategies.STRICT);
+        //map fields from userRequest to userEntity
+        UserEntity userEntity = modelMapper.map(userRequest , UserEntity.class);
+        userEntity.setUserId(UUID.randomUUID().toString());
+        userEntity.setEncryptedPassword(bCryptPasswordEncoder.encode(userRequest.getPassword()));
+        userRepository.save(userEntity);
+
+        UserModel userModel = modelMapper.map(userEntity , UserModel.class);
+        return userModel;
+    }
+
+    @Override
+    public UserEntity getUserDetailsByEmail(String email) throws UsernameNotFoundException {
+        UserEntity userEntity = userRepository.findByEmail(email);
+        if(userEntity == null) {
+            throw new UsernameNotFoundException(email);
```

Comment:
UserNotFoundException maybe ?

Reply:
its UsernameNotFoundException only, recheck once

Reply:
UserNotFound not usernameNotFound. Just to make it more generic